File: src/org/rcgonzalezf/Problem1.java
package org.rcgonzalezf;

/**
 * https://projecteuler.net/problem=1
 * <p>
 * If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
 * <p>
 * Find the sum of all the multiples of 3 or 5 below 1000.
 */
public class Problem1 {

    public int baseNumber = 10;

    public int solution() {
        int solution = 0;

        for (int i = 0; i < baseNumber; ++i) {
            if (i % 3 == 0 || i % 5 == 0) {
                solution += i;
            }
        }

        return solution;
    }
}

File: code/src/main/kotlin/org/rcgonzalezf/onetoten/Problem1.kt
package org.rcgonzalezf.onetoten

/**
 * https://projecteuler.net/problem=1
 *
 *
 * If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
 *
 *
 * Find the sum of all the multiples of 3 or 5 below 1000.
 */
class Problem1 {

    var baseNumber = 10

    fun solution(): Int {
        var solution = 0

        (0..baseNumber - 1).filter { it % 3 == 0 || it % 5 == 0 }.forEach {
            solution += it
        }

        return solution
    }
}
